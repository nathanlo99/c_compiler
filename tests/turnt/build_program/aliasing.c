
int wain(int a, int b) {
  int *c = NULL;
  c = &a;

  a = 69; // Seemingly unused assignment
  b = *c; // Sneakily used here
  a = 70;

  return a + b; // 69 + 70 = 139
}
