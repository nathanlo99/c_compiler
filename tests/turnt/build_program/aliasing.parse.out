procedures main
main INT WAIN LPAREN dcl COMMA dcl RPAREN LBRACE dcls statements RETURN expr SEMI RBRACE
INT int
WAIN wain
LPAREN (
dcl type ID
type INT
INT int
ID a
COMMA ,
dcl type ID
type INT
INT int
ID b
RPAREN )
LBRACE {
dcls dcls dcl BECOMES NULL SEMI
dcls .EMPTY
dcl type ID
type INT STAR
INT int
STAR *
ID c
BECOMES =
NULL NULL
SEMI ;
statements statements statement
statements statements statement
statements statements statement
statements statements statement
statements .EMPTY
statement lvalue BECOMES expr SEMI
lvalue ID
ID c
BECOMES =
expr term
term factor
factor AMP lvalue
AMP &
lvalue ID
ID a
SEMI ;
statement lvalue BECOMES expr SEMI
lvalue ID
ID a
BECOMES =
expr term
term factor
factor NUM
NUM 69
SEMI ;
statement lvalue BECOMES expr SEMI
lvalue ID
ID b
BECOMES =
expr term
term factor
factor STAR factor
STAR *
factor ID
ID c
SEMI ;
statement lvalue BECOMES expr SEMI
lvalue ID
ID a
BECOMES =
expr term
term factor
factor NUM
NUM 70
SEMI ;
RETURN return
expr expr PLUS term
expr term
term factor
factor ID
ID a
PLUS +
term factor
factor ID
ID b
SEMI ;
RBRACE }
